examples/deadletterqueue/movingMessagesToDLQ.ts

```diff
@@ -0,0 +1,192 @@
+  console.log("\n Setup queues, now running scenario B ... ");
+  await processMessageQueue(exceedMaxRetriesProcessor, queuePath);
+}
+
+async function runProcessingErrorHandlingScenario(): Promise<void> {
+  // Prepare given queue by sending few messages
+  await sendMessages();
+
+  // Process messages from queue, by invoking a faulty processor that doesn't successfully process
+  console.log("\n Setup queues, now running scenario C ... ");
+  await processMessageQueue(faultyProcessor, queuePath);
+}
+
+// Helper to send few sample messages to given queue
+async function sendMessages(): Promise<void> {
+  const client = ns.createQueueClient(queuePath);
+  const messageBody = { name: "Apple", color: "Green", qty: 1 };
+  const message: SendableMessageInfo = {
+    body: messageBody,
+    contentType: "application/json",
+    messageId: generateUuid()
+  };
+
+  let i: number;
+  for (i = 0; i < numberOfMessages; i++) {
+    await client.send(message);
+  }
+}
+
+// Helper to process messages by delegating actual processing to a supplied custom messageProcessor
+async function processMessageQueue(messageProcessor: Function, path: string): Promise<void> {
+  const client = ns.createQueueClient(path, { receiveMode: ReceiveMode.peekLock });
+  await messageProcessor(client);
+}
+
+// OnError handler for message receive
+const onError: OnError = (err: MessagingError | Error) => {
+  console.log(">>>>> Error occurred: ", err);
+};
+
+// OnMessage handler for Scenario A - immediate dead lettering of message on receive
+async function simpleDeadletterProcessor(client: QueueClient): Promise<void> {
+  const onMessageHandler: OnMessage = async (brokeredMessage: ServiceBusMessage) => {
+    console.log(">>>>> Killing the message", brokeredMessage);
+    await brokeredMessage.deadLetter();
+  };
+
+  const receiverHandler = await client.receive(onMessageHandler, onError, { autoComplete: false });
+  await delay(receiveClientTimeout);
+  await receiverHandler.stop();
+}
+
+// OnMessage handler for Scenario B - attempts to exceed max allowed retries
+async function exceedMaxRetriesProcessor(client: QueueClient): Promise<void> {
+  const onMessageHandler: OnMessage = async (brokeredMessage: ServiceBusMessage) => {
+    console.log(
+      ">>>>> Attempt to abandon message. Current delivery count: ",
+      brokeredMessage.deliveryCount
+    );
+    await brokeredMessage.abandon();
+  };
+
+  const receiverHandler = await client.receive(onMessageHandler, onError, { autoComplete: false });
+  await delay(receiveClientTimeout);
+  await receiverHandler.stop();
+}
+
+// OnMessage handler for Scenario C - usage for handling reject/fail on message processing
+async function faultyProcessor(client: QueueClient): Promise<void> {
+  const onMessageHandler: OnMessage = async (brokeredMessage: ServiceBusMessage) => {
+    console.log(">>>>> Rejecting the message: ", brokeredMessage);
+    try {
+      await Promise.reject("Green apples are not OK.");
```

Comment:
Is this used in order to trigger and exception?

Reply:
Yes, updated code to throw it as an error.